### ISSUE TITLE
web2's WSGIHandler incorrectly advertises twisted as a multiprocess environment

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1782]
|<img alt="pjenvey's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| pjenvey reported|
|-|-|
|Trac ID|trac#1782|
|Type|defect|
|Created|2006-06-03 05:11:23Z|

web2's WSGIHandler incorrectly sets the 'wsgi.multiprocess' option to True

The WSGI PEP might not explain this in the clearest manner, but this option is reserved for environments that spawn separate processes for separate requests (such as Apache, mod_python)

This incorrect setting leads to WSGI apps making incorrect assumptions about the environment (Paste is a good example -- it disables the EvalException debugging tool if this value is True)

<details><summary>Searchable metadata</summary>

```
trac-id__1782 1782
type__defect defect
reporter__pjenvey pjenvey
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1149311483000000 1149311483000000
changetime__1150567650000000 1150567650000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

The PEP says: 

  This value should evaluate true if an equivalent application object may be simultaneously invoked by another process, and should evaluate false otherwise.

That wording led me to believe it was safest to always set it to true, as it indeed _may_ be invoked by another process. But looking back again, I agree it should actually be False.

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

Sorry for taking so long to get to this. Simplest patch ever, of course. :)

```
Index: wsgi.py
===================================================================
--- wsgi.py     (revision 17257)
+++ wsgi.py     (working copy)
@@ -109,7 +109,7 @@
         env['wsgi.input']        = InputStream(request.stream)
         env['wsgi.errors']       = ErrorStream()
         env['wsgi.multithread']  = True
-        env['wsgi.multiprocess'] = True
+        env['wsgi.multiprocess'] = False
         env['wsgi.run_once']     = False
         env['wsgi.file_wrapper'] = FileWrapper
         self.environment = env
```

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

Apply Apply!

**Comment by twisted-trac** [comment]
|<img alt="foom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|foom set status to `closed`|
|-|-|

(In [17258]) Set wsgi.multiprocess to False rather than True, to better comply with what
the WSGI spec meant to say.

Author: jknight
Reviewer: dreid
Fixes [#1782](https://github.com/twisted/twisted/issues/1782)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
